File: app/src/main/java/com/nytimes/android/external/register/legal/LegalAdapter.java
package com.nytimes.android.external.register.legal;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.nytimes.android.external.register.R;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * simple adapter to bind license items with viewHolder
 */
public class LegalAdapter extends RecyclerView.Adapter<com.nytimes.android.external.register.legal.LegalViewHolder> {

    @NonNull
    private final List<Map.Entry<String, String>> items;
    @NonNull
    private final LayoutInflater inflater;

    LegalAdapter(@NonNull LayoutInflater inflater, @NonNull List<Map.Entry<String, String>> items) {
        this.inflater = inflater;
        this.items = new ArrayList<>(items);
    }

    @Override
    public LegalViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new com.nytimes.android.external.register.legal.LegalViewHolder(
                inflater.inflate(R.layout.item_legal, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull LegalViewHolder holder, int position) {
        holder.onBind(getItem(position));
    }

    Map.Entry<String, String> getItem(int position) {
        return items.get(position);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

}

File: app/src/main/java/com/nytimes/android/external/register/legal/LegalAdapter.kt
package com.nytimes.android.external.register.legal

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.nytimes.android.external.register.R
import java.util.*

/**
 * simple adapter to bind license items with viewHolder
 */
class LegalAdapter(private val inflater: LayoutInflater, items: List<Pair<String, String>>) : RecyclerView.Adapter<LegalViewHolder>() {

    private val items = ArrayList<Pair<String, String>>(items)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LegalViewHolder {
        return LegalViewHolder(
                inflater.inflate(R.layout.nyt_register_item_legal, parent, false))
    }

    override fun onBindViewHolder(holder: LegalViewHolder, position: Int) {
        holder.onBind(getItem(position))
    }

    fun getItem(position: Int): Pair<String, String> {
        return items[position]
    }

    override fun getItemCount(): Int {
        return items.size
    }

}
